### PR TITLE
Utilize Fluent to format numbers and dates in `PDFDocumentProperties`/`AnnotationLayer`

### PR DESCRIPTION
**Pull request by Snuffleupagus**
- **Utilize Fluent to format numbers and dates in `PDFDocumentProperties`**

   The `PDFDocumentProperties` dialog may not display correctly formatted data, especially in the GENERIC viewer, since it's using native methods[1] that depend on the *browser* locale instead of the viewer locale as intended.
   At the time when this dialog was introduced that was probably all we could easily do, but with Fluent we're able to improve things since it's got built-in support for formatting numbers and dates. Not only does this simplify the JavaScript code, but it also gives the localizer more fine-grained control of the desired output.

   Please find additional information here:
    - https://projectfluent.org/fluent/guide/builtins.html
    - https://projectfluent.org/fluent/guide/functions.html

   ---

   [1] `toLocaleString`, `toLocaleDateString`, and `toLocaleTimeString`.

 - **Utilize Fluent to format dates in the `AnnotationLayer`**

   The `AnnotationLayer` may not display correctly formatted data in PopupAnnotations, especially in the GENERIC viewer, since it's using native methods[1] that depend on the *browser* locale instead of the viewer locale as intended.
   With Fluent we're able to improve things since it's got built-in support for formatting dates. Not only does this simplify the JavaScript code slightly, but it also gives the localizer more fine-grained control of the desired output.

   Please find additional information here:
    - https://projectfluent.org/fluent/guide/builtins.html
    - https://projectfluent.org/fluent/guide/functions.html

   ---

   [1] `toLocaleDateString`, and `toLocaleTimeString`.

### CODE REVIEW

File: l10n/sv-SE/viewer.ftl
Comment (flodolo) [inline l10n/sv-SE/viewer.ftl:1]: Please undo these changes, we only touch the en-US file in this repository

**Review by flodolo — APPROVED** [review]
(with the Swedish changes removed)

File: l10n/sv-SE/viewer.ftl
Comment (Snuffleupagus) [reply l10n/sv-SE/viewer.ftl:1]: I've reverted it, even though that's not technically necessary since the l10n-updates we run regularly will simply overwrite any changes made to the non-English `l10n/` files.

**Comment by Snuffleupagus** [comment]
@timvandermeij Mind taking a look at the JS-code in the patch, before we land this?
Note that the test "failures" are expected, since we now format English dates slightly differently than before.

**Review by timvandermeij — APPROVED** [review]

**Comment by timvandermeij** [comment]
Thank you for improving this!

/botio makeref
